Self study and practice for M.Sc. Project

https://towardsdatascience.com/similarity-of-documents-with-random-walks-98f94fd2c76c

In [27]:
import numpy as np


M = np.array([
    [0, 0, 0, 1./3., 0.5],
    [0, 0, 0, 1./3., 0],
    [0, 0, 0, 1./3., 0.5],
    [0.5, 1, 0.5, 0, 0],
    [0.5, 0, 0.5, 0, 0]
])

In [46]:
v = np.array([1., .0, .0, .0, .0]) # starting point
#n=8,20,3 any 
for step in range(4):
    v = np.dot(M, v)

In [47]:
print(v)

[0.40277778 0.19444444 0.40277778 0.         0.        ]


40.3% probability of first place, 19.4% for second, 40.3% for third place